proposals/NNNN-remove-sequence-subsequence.md

```diff
@@ -0,0 +1,327 @@
+
+There is a prototype in [this PR](https://github.com/apple/swift/pull/20175) that replaces the customization points on `Sequence` with regular extensions that each return a specific type.
+
+To see the performance problem, here is how the [benchmarks](https://github.com/apple/swift/pull/20175#issuecomment-434661846) improve if you return a non-type-erased type instead:
+
+#### Performance: -O
+
+TEST                          | OLD  | NEW   | DELTA      | RATIO     
+---                           | ---  | ---   | ---        | ---       
+**Improvement**               |      |       |            |           
+DropWhileSequence             | 2214 | 29    | -98.7%     | **76.34x** 
+PrefixSequenceLazy            | 2265 | 52    | -97.7%     | **43.56x** 
+PrefixSequence                | 2213 | 52    | -97.7%     | **42.56x** 
+DropFirstSequenceLazy         | 2310 | 59    | -97.4%     | **39.15x** 
+DropFirstSequence             | 2240 | 59    | -97.4%     | **37.97x** 
+
+#### Performance: -Osize
+
+TEST                          | OLD   | NEW   | DELTA   | RATIO     
+---                           | ---   | ---   | ---     | ---       
+**Improvement**               |       |       |         |           
+DropWhileAnySeqCRangeIter     | 17631 | 163   | -99.1%  | **108.16x** 
+DropFirstAnySeqCRangeIterLazy | 21259 | 213   | -99.0%  | **99.81x** 
+PrefixAnySeqCRangeIterLazy    | 16679 | 176   | -98.9%  | **94.77x** 
+PrefixAnySeqCntRangeLazy      | 15810 | 168   | -98.9%  | **94.11x** 
+DropFirstAnySeqCntRangeLazy   | 15717 | 213   | -98.6%  | **73.79x** 
+DropWhileSequence             | 2582  | 35    | -98.6%  | **73.77x** 
+DropFirstSequenceLazy         | 2671  | 58    | -97.8%  | **46.05x** 
+DropFirstSequence             | 2649  | 58    | -97.8%  | **45.67x** 
+PrefixSequence                | 2705  | 70    | -97.4%  | **38.64x** 
+PrefixSequenceLazy            | 2670  | 70    | -97.4%  | **38.14x** 
+
+These performance improvements are all down to how well the optimizer can eliminate the wrapper abstractions  when there isn’t the barrier of type erasure in the way. In -Onone builds, you don’t see any speedup.
+
+### How does it block conditional conformance?
+
+The problem with `SubSequence` really became clear when conditional conformance
+was implemented. With conditional conformance, it becomes really important that
+an associated type be able to grow and take on capabilities that line up with
+the capabilities you are adding with each new conformance.
+
+For example, the `Slice` type that is the default `SubSequence` for
+`Collection` grows in capabilities as it’s base grows. So for example, if the
+`Base` is a `RandomAccessCollection`, then so can the `Slice` be. This then
+works nicely when you add new conformances to a `Collection` that _uses_
+`Slice` as it’s `SubSequence` type. For more detail on this, watch Doug’s
+explanation in our WWDC [Swift
+Generics](https://developer.apple.com/videos/play/wwdc2018/406/) talk (starts
```

Comment:
Link directly to the *Conditional Conformances* section?

`(https://developer.apple.com/videos/play/wwdc2018/406/?time=1614)`

Reply:
Huh, I didn't even know you could do links to a specific time, thanks!

Reply:
If you click inside the video's transcript, the address bar will be updated with a timestamp.